vulnfeeds/cvelist2osv/converter.go

```diff
@@ -76,6 +77,67 @@ func extractConversionMetrics(cve cves.CVE5, refs []osvschema.Reference, metrics
 	// TODO(jesslowe): Add more analysis based on ADP containers, CVSS, KEV, CWE, etc.
 }
 
+//attachCWEs extracts and adds CWE IDs from the CVE5 problem-types
+func attachCWEs(v *vulns.Vulnerability, cve cves.CVE5, metrics *ConversionMetrics){
+	foundCWEs := make(map[string]struct{})
+	addCWE := func(id string){
+		id = strings.TrimSpace(id)
+		if id == ""{
+			return
+		}
+		if !strings.HasPrefix(id,"CWE-"){
+			id = "CWE-" + id
+		}
+		foundCWEs[id] = struct{}{}
+	}
+	re := regexp.MustCompile(`CWE-\d+`)
+	
+	// iterate over CNA problem-types
+	for _, pt := range cve.Containers.CNA.ProblemTypes {
+		for _, desc := range pt.Descriptions {
+			// desc is the anonymous struct{Type,Lang,Description string}
+			if desc.Type == "CWE" {
+				if desc.CWEID != ""{
+					addCWE(desc.CWEID)
+				}else{
+					addCWE(desc.Description)
+				}
+			}
+			for _, m := range re.FindAllString(desc.Description, -1) {
+				addCWE(m)
+			}
+		}
+	}
+	// iterate over ADP problem-types
+	for _, adp := range cve.Containers.ADP {
+		for _, pt := range adp.ProblemTypes {
+			for _, desc := range pt.Descriptions {
+				if desc.Type == "CWE" {
+					addCWE(desc.Description) // CWE ID is stored in Description field
+				}
+				for _, m := range re.FindAllString(desc.Description, -1) {
+					addCWE(m)
+				}
+			}
+		}
+	}
+
+	if len(foundCWEs) == 0 {
+		return
+	}
+	list := make([]string, 0, len(foundCWEs))
+	for id := range foundCWEs{
+		list = append(list, id)
+	}
+	slices.Sort(list)
+	if v.DatabaseSpecific == nil{
+		v.DatabaseSpecific = make(map[string]any)
+	}
+	
+	v.DatabaseSpecific["cwes"] = list
```

Comment:
```suggestion
	v.DatabaseSpecific["cwe_ids"] = list
```
nit just to align with some existing GHSA records